README.md

```diff
@@ -5,94 +5,140 @@
 
 
 
-# Description
-Authentication services provided by the xsuaa service on [SAP Cloud Platform](https://cloudplatform.sap.com) or [SAP HANA XS Advanced](https://help.sap.com/viewer/4505d0bdaf4948449b7f7379d24d0f0d/2.0.00/en-US) rely on usage of the [OAuth 2.0](https://oauth.net) protocol and OAuth 2.0 access tokens.
-
-## Web Flow for Authentication
+# SAP BTP Security Services Integration Libraries
+This repository offers a comprehensive set of libraries designed to simplify the integration of [SAP Business Technology Platform](https://www.sap.com/products/technology-platform.html) (SAP BTP) security services (XSUAA and Identity Services).
+Tailored to support Java EE and Spring Boot applications running on Cloud Foundry or Kubernetes environments.
+The libraries focus on streamlining [OAuth 2.0](https://oauth.net) access token validation for tokens issued by XSUAA and Identity Services, in addition, it offers a token-client library to easily fetch tokens without cumbersome setup for http requests. 
+
+## Table of Contents
+1. [Prerequisites](#prerequisites)
+2. [Usage](#usage)
+   - [2.1 Token Validation](#21-token-validation)
+     - [2.1.1 Java EE web applications](#211-Java-EE-web-applications)
+     - [2.1.2 Spring Boot applications](#212-spring-boot-web-applications)
+   - [2.2 Token Flows](#22-token-flows)
+3. [Installation](#installation)
+4. [Troubleshooting](#troubleshooting)
+5. [Contributing](#contributing)
+6. [License](#license)
+
+## Prerequisites
+Before you can use the SAP Cloud Security Services Integration libraries, you must fulfil the following requirements:
+
+1. Access to an SAP BTP account and the XSUAA or identity service.
+2. Familiarity with OAuth 2.0 and JWT (JSON Web Tokens).
+3. Knowledge of Java programming and Spring Boot framework.
+4. Java 17
+5. Maven 3.9.0 or later
+6. (Optional) Spring Boot 3.0.0 or later, Spring Security 6.0.0 or later if using the Spring integration
+
+
+## Usage
 Typical UI5 applications consist of a server providing the HTML content and one or more application serving REST APIs used by the application. Web application use the OAuth Authorization Code Flow for interactive authentication:
```

Comment:
Afaik our customers' clients are not limited to UI5 applications. If this is true, maybe it would be better to generalize this section a bit.